File: backend/src/main/java/org/jdk/project/service/config/LoungeCommandService.java
package org.jdk.project.service.config;

import static org.jdk.project.service.config.ConfigCommandSupport.defaultString;
import static org.jdk.project.service.config.ConfigCommandSupport.ensureUpdated;
import static org.jooq.generated.project.Tables.LOUNGE;

import lombok.RequiredArgsConstructor;
import org.jdk.project.dto.config.LoungeUpsertRequest;
import org.jooq.DSLContext;
import org.jooq.generated.project.tables.pojos.Lounge;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
public class LoungeCommandService {

  private final DSLContext dsl;

  @Transactional
  public Long create(LoungeUpsertRequest request) {
    Lounge lounge = new Lounge();
    lounge.setCode(request.getRoomCode());
    lounge.setName(request.getDeptName());
    lounge.setTerminal(defaultString(request.getTerminal(), ""));
    lounge.setLocationDesc(defaultString(request.getLocationDesc(), ""));
    lounge.setEnabled(request.getEnabled() == null || request.getEnabled());
    lounge.setRemark(defaultString(request.getRemark(), ""));
    return dsl.insertInto(LOUNGE)
        .set(dsl.newRecord(LOUNGE, lounge))
        .returningResult(LOUNGE.ID)
        .fetchOne(LOUNGE.ID);
  }

  @Transactional
  public void update(Long id, LoungeUpsertRequest request) {
    int updated =
        dsl.update(LOUNGE)
            .set(LOUNGE.CODE, request.getRoomCode())
            .set(LOUNGE.NAME, request.getDeptName())
            .set(LOUNGE.TERMINAL, defaultString(request.getTerminal(), ""))
            .set(LOUNGE.LOCATION_DESC, defaultString(request.getLocationDesc(), ""))
            .set(LOUNGE.ENABLED, request.getEnabled() == null || request.getEnabled())
            .set(LOUNGE.REMARK, defaultString(request.getRemark(), ""))
            .where(LOUNGE.ID.eq(id))
            .execute();
    ensureUpdated(updated, "贵宾室不存在");
  }

  @Transactional
  public void delete(Long id) {
    dsl.deleteFrom(LOUNGE).where(LOUNGE.ID.eq(id)).execute();
  }
}

File: backend/src/main/kotlin/org/jdk/project/service/config/LoungeCommandService.kt
package org.jdk.project.service.config

import org.jdk.project.dto.config.LoungeUpsertRequest
import org.jdk.project.service.config.ConfigCommandSupport.defaultString
import org.jdk.project.service.config.ConfigCommandSupport.ensureUpdated
import org.jooq.DSLContext
import org.jooq.generated.project.Tables.LOUNGE
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class LoungeCommandService(
    private val dsl: DSLContext,
) {
    @Transactional
    fun create(request: LoungeUpsertRequest): Long? =
        dsl.insertInto(LOUNGE)
            .set(LOUNGE.CODE, request.roomCode)
            .set(LOUNGE.NAME, request.deptName)
            .set(LOUNGE.TERMINAL, defaultString(request.terminal, ""))
            .set(LOUNGE.LOCATION_DESC, defaultString(request.locationDesc, ""))
            .set(LOUNGE.ENABLED, request.enabled ?: true)
            .set(LOUNGE.REMARK, defaultString(request.remark, ""))
            .returningResult(LOUNGE.ID)
            .fetchOne(LOUNGE.ID)

    @Transactional
    fun update(id: Long, request: LoungeUpsertRequest) {
        val updated =
            dsl.update(LOUNGE)
                .set(LOUNGE.CODE, request.roomCode)
                .set(LOUNGE.NAME, request.deptName)
                .set(LOUNGE.TERMINAL, defaultString(request.terminal, ""))
                .set(LOUNGE.LOCATION_DESC, defaultString(request.locationDesc, ""))
                .set(LOUNGE.ENABLED, request.enabled ?: true)
                .set(LOUNGE.REMARK, defaultString(request.remark, ""))
                .where(LOUNGE.ID.eq(id))
                .execute()
        ensureUpdated(updated, "贵宾室不存在")
    }

    @Transactional
    fun delete(id: Long) {
        dsl.deleteFrom(LOUNGE).where(LOUNGE.ID.eq(id)).execute()
    }
}
